packages/lib-utils/src/app/redux/reducers/k8s/index.ts

```diff
@@ -1,15 +1,271 @@
+import { consoleLogger } from '@monorepo/common';
 import { Map as ImmutableMap, fromJS } from 'immutable';
-import type { ActionType as Action } from 'typesafe-actions';
-import type { K8sModelCommon } from '../../../../types/k8s';
+import * as _ from 'lodash-es';
+import { getReferenceForModel, getNamespacedResources, allModels } from '../../../../k8s/k8s-utils';
+import type { K8sModelCommon, K8sResourceCommon } from '../../../../types/k8s';
 import type { K8sState } from '../../../../types/redux';
+import type { K8sAction } from '../../actions/k8s';
+import { ActionType } from '../../actions/k8s';
+import { getK8sDataById } from './selector';
 
-// TODO remove when actions/reducers are migrated
-const emptyK8sState: K8sState = fromJS({
-  RESOURCES: { inFlight: false, models: ImmutableMap<string, K8sModelCommon>() },
-});
+const getQN: (obj: K8sResourceCommon) => string = (obj: K8sResourceCommon) => {
+  const { name, namespace } = obj?.metadata || {};
+  // Name + namespace is not unique for PackageManifest resources, so include the catalog source.
+  // TODO: We should be able to remove this when the upstream OLM bug is fixed:
+  //       https://bugzilla.redhat.com/show_bug.cgi?id=1814822
+  if (obj.apiVersion === 'packages.operators.coreos.com/v1' && obj.kind === 'PackageManifest') {
+    return `(${obj.status?.catalogSource})-${name}`;
+  }
+  return (namespace ? `(${namespace})-` : '') + name;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const moreRecent = (a: any, b: any) => {
+  const metaA = a.get('metadata').toJSON();
+  const metaB = b.get('metadata').toJSON();
+  if (metaA.uid !== metaB.uid) {
+    return new Date(metaA.creationTimestamp) > new Date(metaB.creationTimestamp);
+  }
+  return parseInt(metaA.resourceVersion, 10) > parseInt(metaB.resourceVersion, 10);
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const removeFromList = (list: ImmutableMap<string, any>, resource: K8sResourceCommon) => {
```

Comment:
```suggestion
const removeFromList = (list: ImmutableMap<string, unknown>, resource: K8sResourceCommon) => {
```

Don't think we care what the value is... unless we get a centralized type for this item, maybe `unknown` is all we should do.